server/src/main/java/com/coincoinche/engine/BiddingGameState.java

```diff
@@ -0,0 +1,64 @@
+package com.coincoinche.engine;
+
+import com.coincoinche.engine.teams.Player;
+import java.util.ArrayList;
+import java.util.List;
+
+/** TODO nockty add detailed documentation here. */
+public class BiddingGameState implements GameStateInterface {
+
+  private Player currentPlayer;
+  private Contract highestBidding;
+  private boolean coinched;
+
+  BiddingGameState(
+      Player currentPlayer, Contract highestBidding, boolean coinched, boolean surcoinched) {
```

Comment:
I think it is useful to have a constructor with these arguments because:
- we can build an arbitrary state for testing purposes (note that the constructor is not public, it is only visible inside the package)
- it may be useful if we want to build an ongoing game, for instance after the server has restarted (we re-build a persisted game)